template/src/lib/sdk/google/GoogleMap.js

```diff
@@ -43,31 +46,45 @@ export function GoogleMap(props) {
 
       const handleViewportChangeDebounced = debounce(handleViewportChange, 200);
       map.addListener('bounds_changed', handleViewportChangeDebounced);
+      map.addListener('resize', () => {
+        props.onResize({
+          height: map.getDiv().offsetHeight,
+          width: map.getDiv().offsetWidth,
+        });
+      });
+
+      window.onresize = () => {
+        triggerResize(map);
+      };
+
+      triggerResize(map);
 
-      // keep references for further updates
       window.cartoGmap = map;
       window.cartoDeck = deckOverlay;
     } else {
       const { center, zoom, ...rest } = options;
       window.cartoGmap.setZoom(zoom);
       window.cartoGmap.setCenter(center);
       window.cartoGmap.setOptions(rest);
-      window.cartoDeck.setProps({ layers });
+      window.cartoDeck.setProps({ layers, getTooltip });
     }
 
     window.cartoDeck.setMap(window.cartoGmap);
   };
 
   useEffect(() => {
-    if (!window.google) {
+    if (!document.querySelector('#gmaps')) {
```

Comment:
I first thought that this id could identify the 'div' for the map. Then I have seen that you assign the id to the script, so I guess that's ok. I maybe would use a bit longer name, to avoid a collision with an hypothetical 'gmaps' div for the map